### PR TITLE
OCaml 5.0 compatibility

### LINKED ISSUE #11271: OCaml 5 compat
Fedora decided to upgrade OCaml to 5.x in their distro, thus we need to sort out OCaml 5 compat. A partial patch was provided in https://bugzilla.redhat.com/show_bug.cgi?id=2218692

Many other distros, and older versions of them, has only OCaml 4, so we need to support both OCaml 4 and 5 in some way.

If using a single code base is not possible to support both OCaml 4 and 5, we can maintain a seperate branch, such that I can generate patches from that branch for packaging purpose.

### PR DESCRIPTION
**Pull request by kLabz**
Makes the codebase compatible with both ocaml 4.08 and 5.0. CI now runs for both on linux.

I had to disable StdThread.kill on eval which was probably not even working in the first place. Will likely need to disable it more nicely (from std side?) and cleanup the commented out code.

Closes #11271

### CODE REVIEW

**Comment by andyli** [comment]
Would you rebase it on top of 4.3.2 such that I can use it as a patch for packaging?

**Comment by kLabz** [comment]
I can't until I have a new implementation for the memory-related features :/

**Comment by andyli** [comment]
I thought you've disabled the memory features temporily in this PR?
It would be nice to have 4.3.2 packages even if the memory features are disabled (I guess they're not critical).
To be clear, I'm not asking for merging into development, but keeping a seperate ocaml-5-ready branch/PR for the 4.3_bugfix branch.

**Comment by kLabz** [comment]
I updated the lib we needed for memory features so that it works on both ocaml 4.08 and 5.0; I didn't want 4.3.x releases missing features that would break some editor features.

I'm not sure what exactly you need. If this PR produces what you're after, I can backport it to 4.3_bugfix branch tomorrow morning.

**Comment by andyli** [comment]
Great! Thanks a lot for the quick work.

> backport it to 4.3_bugfix branch

Yes, that will do. I will than able to generate a patch to apply it on top of the 4.3.2 source.

File: haxe.opam [hunk 1]
```diff
@@ -19,14 +19,15 @@ build: [
 install: [make "install" "INSTALL_DIR=%{prefix}%"]
 remove: [make "uninstall" "INSTALL_DIR=%{prefix}%"]
 depends: [
-  "ocaml" {>= "4.08"}
+  ("ocaml" {>= "5.0"} & "camlp5")
```
Comment (andyli) [inline haxe.opam]: shouldn't camlp5 still a build-only dep? i.e. `("ocaml" {>= "5.0"} & ("camlp5" {build}))`

**Comment by kLabz** [comment]
I also pushed to [`4.3_bugfix`](https://github.com/HaxeFoundation/haxe/commits/4.3_bugfix) (after checking CI on a separate branch), you'll have to wait a couple hours for CI to complete (arm64.. :/) if you need it otherwise you should be good to go.

**Pull request merged**